log-collector/src/main/scala/io/phdata/pulse/logcollector/LogCollector.scala

```diff
@@ -94,7 +94,10 @@ object LogCollector extends LazyLogging {
 
     httpServerFuture.onComplete {
       case Success(v) => ()
-      case Failure(e) => throw new RuntimeException(e)
+      case Failure(ex) => {
+        logger.error("Exception caught ", ex)
```

Comment:
This message doesn't add much, how about a more specific failure like 'HTTP server failed, exiting.'